File: flink-java-code/src/main/java/com/atguigu/day2/SensorStream.java
package com.atguigu.day2;

import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

public class SensorStream {
    public static void main(String[] args) throws Exception {
        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);

        DataStream<SensorReading> stream = env.addSource(new SensorSource());

        stream.print();

        env.execute();
    }
}

File: flink-scala-code/src/main/scala/com/atguigu/day2/SensorStream.scala
package com.atguigu.day2

import org.apache.flink.streaming.api.scala._

object SensorStream {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(1)

    val stream = env.addSource(new SensorSource)

    stream.print()

    env.execute()
  }
}